### ISSUE TITLE
[Flow Item] Inset border on back button for 4.0 (Feb)

### ISSUE THREAD

**Issue by DitwanP** [Esri/calcite-design-system#10781]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

Change the back button border to be inset.

[Figma specs](https://www.figma.com/design/5nE4ZBy9mKdYjGmMZmftTP/%5BFlow%5D-Inset-border-on-back-button?node-id=301-305&node-type=canvas&t=uBwE1QQ5Gcwb1xfs-0)

### Acceptance Criteria

- [ ] Inset the border on back button

### Relevant Info

Part of 4.0's Phase 2

Blocked by: #10759

### Which Component

- `flow-item`

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Calcite (design)

**Comment by ashetland** [comment]
Closing in favor of #10777. Updates to the back button used in Flow Item are documented there.
